### DISCUSSION TITLE
Build a dedicated mobile app ?

### DISCUSSION THREAD

**Discussion by raphodn** [openfoodfacts/open-prices#570] [Ideas]
Current mobile app: https://github.com/openfoodfacts/smooth-app

### Why ?

- better barcode scanner (than in the web app)
- dedicated pages to visualize price data
- dedicated workflows to contribute data
- user personal dashboard, basket price tracking, basket Nutriscore/CO2 impact...
- get new users outside of the OFF community

**Comment by serpico** [comment]
Sometimes I'd like to, mostly, be able to track the "life cycle" of products I bought a few examples :

- When I buy products in bulk and would like to see how long it'll last ( when bought, when open, when finished ) until it's finished - also to help me remember when I need to restock and not run out ), same goes for products.
- For products I deem not very healthy I'd like to monitor my consumption ( I my mind it's ok if I consume them slowly over a few months - I can manage the craving this way as well ).
- For products with high carbon footprint sometimes I don't buy them but I "log" them into open price and I'd like to know how much CO2 I "saved" over a period of times ( intentionally not buying them for a few weeks/months ).

**Reply by raphodn** [reply]
> For products with high carbon footprint sometimes I don't buy them but I "log" them into open price and I'd like to know how much CO2 I "saved" over a period of times

coincidence : stumbled on [this](https://france3-regions.francetvinfo.fr/auvergne-rhone-alpes/isere/grenoble/environnement-un-grenoblois-propose-d-afficher-le-bilan-carbone-des-produits-sur-les-tickets-de-caisse-2578504.html) today

**Reply by serpico** [reply]
👀
![concombre+5%](https://github.com/user-attachments/assets/b6801505-d27e-4d38-934a-2f7840668674)

Me : frantically goes to https://leprixdescarottes.fr/ to check how much those 5% would set me back

**Comment by raphael0202** [comment]
We discussed yesterday this issue with @g123k, and while having a dedicated Open Prices mobile app seems very desirable in the mid-run, it's not something we will be able to do any time soon.
Edouard talked about better integrating Prices in the Open Food Facts app so that we have more contribution.

**Comment by TTalex** [comment]
Is there a way to gather stats on the progressive web app usage ?
This is what I'm using on my phone, and it pretty much behaves like a regular app. So in my case, I do not really see huge benefits for a native app ?

**Reply by raphael0202** [reply]
>  This is what I'm using on my phone, and it pretty much behaves like a regular app. So in my case, I do not really see huge benefits for a native app ?

A good barcode scanner is the main advantage of the mobile app for now! For the rest the web app is better imo.

**Reply by teolemon** [reply]
We have matomo @TTalex 
we could wire more precise events @raphael0202 @raphodn

**Reply by raphodn** [reply]
What kind of usage ? The share of users on mobile vs web ?

**Reply by TTalex** [reply]
Ideally, it would be interesting to know how many people have "installed" the progressive web app. I know operating systems aren't consistent on offering the installation option, so I'm not expecting a high number.
